Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -582,6 +582,13 @@ private void RetestSubmission(DatabaseModelType submission)
                 return;
             }
 
+            var submissionForProcessing = this.Data.SubmissionsForProcessing.GetById(submission.Id);
```

Comment:
Shouldn't  you use another method to get the submission like GetBySubmissionId, unless GetById is overriden it uses a default implementation that gets a submission by it's primary key Id (which for the SubmissionsForProcessing table is not the Submission id from what I understand).